lib/iaas/AzureClient.js

```diff
@@ -0,0 +1,192 @@
+'use strict';
+const _ = require('lodash');
+const Promise = require('bluebird');
+const azureStorage = require('azure-storage');
+const logger = require('../logger');
+const errors = require('../errors');
+const utils = require('../utils');
+const ComputeClient = require('./ComputeClient');
+const CONST = require('../constants');
+const BaseCloudClient = require('./BaseCloudClient');
+const NotFound = errors.NotFound;
+const Unauthorized = errors.Unauthorized;
+const Forbidden = errors.Forbidden;
+
+class AzureClient extends BaseCloudClient {
+  constructor(settings) {
+    super(settings);
+    this.storage = this.constructor.createStorageClient(_
+      .chain(this.settings)
+      .omit('name')
+      .set('provider', this.provider)
+      .value()
+    );
+    this.computeClient = this.constructor.createComputeClient(_
+      .chain(this.settings)
+      .set('provider', this.provider)
+      .value()
+    );
+    Promise.promisifyAll(this.computeClient.snapshots);
+
+    this.storage.on('log::*', function onmessage(message, obj) {
+      const event = _.nth(this.event.split('::'), 1);
```

Comment:
what is this.event? shouldn't you be splitting the message instead?

Reply:
this should be based on event only. If you see next statement... it is looking for log level. But the problem is I inherited the code from CloudProviderClient.js which uses package cloud. But I am using azure storage lib which also has event listener. But I didn't see any example how to use this. Planning to remove this listener.